src/main/java/Duke.java

```diff
@@ -23,19 +25,38 @@ public static void main(String[] args) {
             }
             else if (userInput.equals("list")) {
                 System.out.println("-----------------------------");
-                for (int i = 0; i < taskList.size(); i++) {
-                    //System.out.println(45+5 + "=" +45+5)
-                    System.out.printf(i+1 + ". ");
-                    System.out.println(taskList.get(i) + " ");
+                System.out.println("Olivia presents you a list of tasks for you to do:");
+                for (Task task : taskList) {
+                    System.out.print(task.getTaskId()+".");
+                    System.out.print("["+task.getTaskStatus()+"] ");
+                    System.out.println(task.getDescription());
+                }
+                System.out.println("-----------------------------");
+            }
+            else if (userInput.contains("mark")) {
+                System.out.println("-----------------------------");
+                String[] arraySplit = userInput.split(" "); //determine raw value of task number to mark/unmark
+                int indexAlter = Integer.valueOf(arraySplit[1]) -1; //index of description to mark/unmark
+                if (userInput.contains("un")) {
+                    taskList.get(indexAlter).markAsNotDone();
+                    System.out.println("Hey!!! Please get the task done.");
+                    System.out.println("    [" + taskList.get(indexAlter).getTaskStatus() + "] " + taskList.get(indexAlter).getDescription());
+                }
+                else {
+                    taskList.get(indexAlter).markAsDone();
+                    System.out.println("Olivia thanks you for completing the task! :)");
+                    System.out.println("    [" + taskList.get(indexAlter).getTaskStatus() + "] " + taskList.get(indexAlter).getDescription());
                 }
                 System.out.println("-----------------------------");
             }
             else {
-                taskList.add(userInput);
+                Task t = new Task(userInput); //create new task
+                taskList.add(t); //append task to arraylist
                 System.out.println("-----------------------------");
                 System.out.print("added: ");
                 System.out.println(userInput);
                 System.out.println("-----------------------------");
+
             }
         } while(!userInput.equals("bye"));
```

Comment:
Would it be a problem of using do...while statement in the Duke? Consider a scenario when you input nothing but "bye", it will record it into the list before checking it equals "bye". It might be better if using `while(!userInput.equals(“bye”))` statement